### PR TITLE
new(tests): EOF - EIP-7480: Add DATACOPY edge cases for common int sizes

### PR DESCRIPTION
**Pull request by shemnon**
## 🗒️ Description

Add tests for the DATACOPY data offset, to check for cases where trimming the u256 to a more developer friendly size (u64 / long for example) results in erroneous indexes into the data section.

## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def create_data_test(offset: int, datasize: int):
         pytest.param(0x5BFE, 0x5C00, id="short_23k"),
         pytest.param(0x5BE0, 0x5D00, id="exact_23k"),
         pytest.param(0x2345, 0x5C00, id="large_23k"),
+        pytest.param(2**16-1, 32, id="u16_max"),
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py:84]:
```suggestion
        pytest.param(2**16 - 1, 32, id="u16_max"),
```
Tox complains about no spaces between plus sign.

File: tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py [hunk 2]
```diff
@@ -81,6 +81,12 @@ def create_data_test(offset: int, datasize: int):
...
+        pytest.param(2**16, 32, id="u16_max_plus_1"),
+        pytest.param(2**32-1, 32, id="u32_max"),
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py:86]:
```suggestion
        pytest.param(2**32 - 1, 32, id="u32_max"),
```

File: tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py [hunk 3]
```diff
@@ -81,6 +81,12 @@ def create_data_test(offset: int, datasize: int):
...
+        pytest.param(2**32, 32, id="u32_max_plus_1"),
+        pytest.param(2**64-1, 32, id="u64_max"),
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip7480_data_section/test_data_opcodes.py:88]:
```suggestion
        pytest.param(2**64 - 1, 32, id="u64_max"),
```

**Review by marioevz — APPROVED** [review]
LGTM, thanks!
